lib/Ajax/Application/Handler/SwitchEncryption.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * Copyright 2012-2017 Horde LLC (http://www.horde.org/)
+ *
+ * See the enclosed file LICENSE for license information (GPL). If you
+ * did not receive this file, see http://www.horde.org/licenses/gpl.
+ *
+ * @category  Horde
+ * @copyright 2012-2017 Horde LLC
+ * @license   http://www.horde.org/licenses/gpl GPL
+ * @package   IMP
+ */
+
+/**
+ * Defines AJAX actions used in the IMP alias dialog.
+ *
+ * @author    Rafael te Boekhorst <boekhorst@b1-systems.de>
+ * @author    Michael Slusarz <slusarz@horde.org>
+ * @category  Horde
+ * @copyright 2012-2017 Horde LLC
+ * @license   http://www.horde.org/licenses/gpl GPL
+ * @package   IMP
+ */
+class IMP_Ajax_Application_Handler_SwitchEncryption extends Horde_Core_Ajax_Application_Handler
+{
+    /**
+     * AJAX action: Get Keys from Identity of adressbook.
+     *
+     * @return array  Array with keys of identity.
+     */
+    public function getIdentityPubKey()
+    {
+        global $injector, $notification;
+
+        // get ID of identity and email
+        $identityID = $this->vars->identityID;
+        $identity = $injector->getInstance('IMP_Identity');
+        $email = $identity->getEmail($identityID);
+        $name = $identity->getName($identityID);
+
+        $smime_url = IMP_Basic_Smime::url();
+        $self_url = Horde::url($GLOBALS['registry']->getServiceLink('prefs', 'imp'), false);
```

Comment:
When factoring out, use the globals (registry, injector, notification) either as parameters passed in or use their results. We don't want globals magic inside logic.

Reply:
Ok, Im trying to do that. 
So is [this](https://github.com/maintaina-com/imp/pull/9/commits/a74fe46dc035a1c1c126079a8b5da998f6175b3a) what you mean more or less? Added injector as a parameter and called all other classes by `use class...` @ralflang ?